lib/src/features/leaderboard/ui/leaderboard_screen.dart

```diff
@@ -0,0 +1,216 @@
+                        : SliverList(
+                            delegate:
+                                SliverChildListDelegate(_buildList(context)),
+                          )))
+          ]);
+        }));
+  }
+
+  Widget _buildAndroid(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(
+        title: Text(context.l10n.leaderboard),
+      ),
+      body: LayoutBuilder(
+        builder: (BuildContext context, BoxConstraints constraints) {
+          if (constraints.maxWidth > 600) {
+            return GridView(
+              shrinkWrap: true,
+              gridDelegate: SliverGridDelegateWithFixedCrossAxisCount(
+                mainAxisExtent: 644,
+                crossAxisCount: (constraints.maxWidth / 300).floor(),
+              ),
+              children: _buildList(context),
+            );
+          } else {
+            return ListView(
+              children: _buildList(context),
+            );
+          }
+        },
+      ),
+    );
+  }
+
+  List<Widget> _buildList(BuildContext context) {
+    return [
+      _buildLeaderboard(
+          context, leaderboard.bullet, LichessIcons.bullet, 'BULLET'),
+      _buildLeaderboard(
+          context, leaderboard.blitz, LichessIcons.blitz, 'BLITZ'),
+      _buildLeaderboard(
+          context, leaderboard.rapid, LichessIcons.rapid, 'RAPID'),
+      _buildLeaderboard(
+          context, leaderboard.classical, LichessIcons.classical, 'CLASSICAL'),
+      _buildLeaderboard(context, leaderboard.ultrabullet,
+          LichessIcons.ultrabullet, 'ULTRA BULLET'),
+      _buildLeaderboard(
+          context, leaderboard.crazyhouse, LichessIcons.h_square, 'CRAZYHOUSE'),
+      _buildLeaderboard(
+          context, leaderboard.chess960, LichessIcons.die_six, 'CHESS 960'),
+      _buildLeaderboard(context, leaderboard.kingOfThehill, LichessIcons.bullet,
+          'KING OF THE HILL'),
+      _buildLeaderboard(context, leaderboard.threeCheck,
+          LichessIcons.three_check, 'THREE CHECK'),
+      _buildLeaderboard(
+          context, leaderboard.atomic, LichessIcons.atom, 'ATOMIC'),
+      _buildLeaderboard(
+          context, leaderboard.horde, LichessIcons.horde, 'HORDE'),
+      _buildLeaderboard(
+          context, leaderboard.antichess, LichessIcons.antichess, 'ANTICHESS'),
+      _buildLeaderboard(context, leaderboard.racingKings,
+          LichessIcons.racing_kings, 'RACING KINGS'),
+    ];
+  }
+
+  Widget _buildLeaderboard(BuildContext context, List<LightUser> userList,
+      IconData iconData, String title) {
+    return Column(
+      mainAxisSize: MainAxisSize.min,
+      children: [
+        ListTile(
+          leading: Icon(iconData, color: LichessColors.brag),
+          title: Text(title),
+        ),
+        const Divider(),
+        ..._leaderboardList(context, userList),
+        const SizedBox(height: 12),
+      ],
+    );
+  }
+
+  List<Widget> _leaderboardList(
```

Comment:
I'd prefer not to have that function and write the code directly inside `_buildLeaderboard`